bson/src/main/org/bson/codecs/IterableCodec.java

```diff
@@ -17,25 +17,26 @@
 package org.bson.codecs;
 
 import org.bson.BsonReader;
-import org.bson.BsonType;
 import org.bson.BsonWriter;
 import org.bson.Transformer;
 import org.bson.UuidRepresentation;
+import org.bson.codecs.configuration.CodecConfigurationException;
 import org.bson.codecs.configuration.CodecRegistry;
 
-import java.util.ArrayList;
+import java.lang.reflect.Type;
 import java.util.List;
 
 import static org.bson.assertions.Assertions.notNull;
-import static org.bson.codecs.ContainerCodecHelper.readValue;
+import static org.bson.codecs.ContainerCodecHelper.getCodec;
 
 /**
  * Encodes and decodes {@code Iterable} objects.
  *
  * @since 3.3
  */
 @SuppressWarnings("rawtypes")
-public class IterableCodec implements Codec<Iterable>, OverridableUuidRepresentationCodec<Iterable> {
+public class IterableCodec extends AbstractIterableCodec<Object>
```

Comment:
`IterableCodec` now implements `Codec<Iterable<Object>>` instead of `Codec<Iterable>`.  This is binary but not source compatibile, which I think is ok given how this class is likely (not) used directly in applications.